From f45fde63e5bddd4620e88845c94eedf6e527ac4b Mon Sep 17 00:00:00 2001
From: Bogdan Purcareata <bogdan.purcareata@nxp.com>
Date: Fri, 13 Oct 2017 15:45:43 +0000
Subject: [PATCH 128/767] staging: fsl-dpaa2/eth: Update Rx headroom
 computation

Instead of the old opaque value, define the Rx buffer headroom size
based on other frame buffer fields, in accordance to how we expect them
to be configured by MC.

Based on DPNI_BUF_LAYOUT_OPT_TIMESTAMP, MC will enable an additional
field in the Tx hardware annotation area, increasing its size to 128.
Update the driver value to reflect this.

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
index 27bd80aa4d9a..48f815f726c8 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
@@ -121,6 +121,10 @@
  */
 #define DPAA2_ETH_SWA_SIZE		64
 
+/* Hardware annotation area in RX / TX  buffers */
+#define DPAA2_ETH_RX_HWA_SIZE		64
+#define DPAA2_ETH_TX_HWA_SIZE		128
+ 
 /* Must keep this struct smaller than DPAA2_ETH_SWA_SIZE */
 struct dpaa2_eth_swa {
 	struct sk_buff *skb;
-- 
2.17.0

